packages/main/src/plugin/docker/docker-context-handler.spec.ts

```diff
@@ -0,0 +1,202 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as fs from 'node:fs';
+import { homedir } from 'node:os';
+import { join } from 'node:path';
+
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+
+import type { DockerContextParsingInfo } from './docker-context-handler.js';
+import { DockerContextHandler } from './docker-context-handler.js';
+
+export class TestDockerContextHandler extends DockerContextHandler {
+  override getDockerConfigPath(): string {
+    return super.getDockerConfigPath();
+  }
+
+  override getCurrentContext(): Promise<string> {
+    return super.getCurrentContext();
+  }
+
+  override getContexts(): Promise<DockerContextParsingInfo[]> {
+    return super.getContexts();
+  }
+}
+
+// mock exists sync
+vi.mock('node:fs');
+
+const originalConsoleError = console.error;
+let dockerContextHandler: TestDockerContextHandler;
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  console.error = vi.fn();
+  dockerContextHandler = new TestDockerContextHandler();
+});
+
+afterEach(() => {
+  console.error = originalConsoleError;
+});
+
+test('getDockerConfigPath', async () => {
+  const configPath = dockerContextHandler.getDockerConfigPath();
+  expect(configPath).toEqual(join(homedir(), '.docker', 'config.json'));
+});
+
+describe('getCurrentContext', () => {
+  test('should return default if docker config does not exist', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(false);
+
+    const currentContext = await dockerContextHandler.getCurrentContext();
+    expect(currentContext).toBe('default');
+  });
+
+  test('should return context if docker config exist', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(true);
+    vi.spyOn(fs.promises, 'readFile').mockResolvedValue(JSON.stringify({ currentContext: 'test-context' }));
+
+    const currentContext = await dockerContextHandler.getCurrentContext();
+    expect(currentContext).toBe('test-context');
+  });
+
+  test('should return undefined if docker config exist but fails to parse it', async () => {
```

Comment:
Says 'undefined' but returns 'default', I think test title just needs to be updated.